packages/common/src/migrations/upgrade-two-dot-six.ts

```diff
@@ -0,0 +1,109 @@
+/** @license
+ * Copyright 2018 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ISolutionItem, IItemTemplate, UserSession } from "../interfaces";
+import { getProp, cloneObject, cleanItemId } from "../generalHelpers";
+import { getPortal } from "../restHelpersGet";
+import { searchGroups, searchGroupContents } from "../restHelpers";
+
+/**
+ * Hub creates a Web Map dependency for Form templates
+ * that contain map questions. This migration updates
+ * any form's map questions to use the default basemap
+ * and filters out related Web Map templates.
+ * @param {ISolutionItem} model A Solution model
+ * @param {UserSession} authentication The user session info
+ * @returns {Promise<ISolutionItem>}
+ */
+export function _upgradeTwoDotSix(
+  model: ISolutionItem,
+  authentication: UserSession
+): Promise<ISolutionItem> {
+  if (getProp(model, "item.properties.schemaVersion") >= 2.6) {
+    return Promise.resolve(model);
+  } else {
+    const clone: ISolutionItem = cloneObject(model);
+    const formTemplates = clone.data.templates.filter(
+      ({ type }) => type === "Form"
+    );
+    clone.item.properties.schemaVersion = 2.6;
+
+    if (!formTemplates.length) {
+      return Promise.resolve(clone);
+    }
+
+    // fetch org details
+    return getPortal("", authentication)
+      .then(portal => {
```

Comment:
This seems like stuff we should be doing as part of the deployment process as it is customizing the template for the current user's org. So if we created a template via Solution.js, it gets schemaVersion 3... how does that have the basemap stuff for the deploying user vs the person who created the template?

Reply:
> This seems like stuff we should be doing as part of the deployment process as it is customizing the template for the current user's org. So if we created a template via Solution.js, it gets schemaVersion 3... how does that have the basemap stuff for the deploying user vs the person who created the template?

Form templates with map questions created by Solutions.js (schemaVersion 3), reference the same base map item (which is an esri-owned, publicly-accessible base map) as the source Form item & do not create a Web Map template for the basemap.

Form templates with map questions created by Hub create a separate Web Map template for the basemap, which @brollison wanted to avoid, so the decision was made to filter out the basemap templates and default the map questions to use the org default.

In hindsight, I agree, the actual overriding of the basemap probably makes more sense downstream in the type specific processing. Will refactor this PR to:

1) Migration will remain, but it will only filter the webmap templates
2) Org default basemap will be overridden downstream in the type specific processor

Reply:
> 1. Migration will remain, but it will only filter the webmap templates
> 2. Org default basemap will be overridden downstream in the type specific processor

@dbouwman @MikeTschudi 

Updated per my above comment, ready for re-review